kratos/utilities/math_utils.h

```diff
@@ -818,37 +818,40 @@ class MathUtils
  
 	return cross;
     }
-    
+
     /**
-     * Identical but it assumes that the output vector is given already the correct size
-     * @param a: First input vector
-     * @param b: Second input vector
-     * @return c: The resulting vector
+     * Performs the cross product of the two input vectors a,b
+     * a,b are assumed to be of size 3 (check is only performed on vector sizes in debug mode)
+     * @param a First input vector
+     * @param b Second input vector
+     * @param c The resulting vector
      */
     
-    static inline void CrossProduct(
-        array_1d<double, 3>& c, 
-        const array_1d<double, 3>& a, 
-        const array_1d<double, 3>& b
-        )
-    {
-        c[0] = a[1]*b[2] - a[2]*b[1];
-        c[1] = a[2]*b[0] - a[0]*b[2];
-        c[2] = a[0]*b[1] - a[1]*b[0];
-    }
-    
-    //general version    
-    template< class T1, class T2, class T3 >
+    template< class T1, class T2 , class T3>
     static inline void CrossProduct(T1& c, const T2& a, const T3& b ){
+#ifdef KRATOS_DEBUG
+        if (a.size() != 3 || b.size() != 3) 
+        {
+            KRATOS_ERROR << "The size of the vectors is different of 3: " << a << " and " << b << std::endl;
+        }
+#endif
         c[0] = a[1]*b[2] - a[2]*b[1];
         c[1] = a[2]*b[0] - a[0]*b[2];
         c[2] = a[0]*b[1] - a[1]*b[0];
     }
 
-    template< class T1, class T2, class T3 >
+    /**
+     * Performs the unitary cross product of the two input vectors a,b
+     * a,b are assumed to be of size 3 (no check is performed on vector sizes)
+     * @param a First input vector
+     * @param b Second input vector
+     * @param c The resulting vector
+     */
+    
+    template< class T1, class T2 , class T3>
     static inline void UnitCrossProduct(T1& c, const T2& a, const T3& b ){
         CrossProduct(c,a,b);
-        double norm = norm_2(c);
+        const double norm = norm_2(c);
```

Comment:
make the same size check as for the CrossProduct if you are in idebug

Reply:
No, because this calls the CrossProduct, so it is done there

Reply:
you are right